float64 pose_x
float64 pose_y
float64 force_x
float64 force_y